felix/dataplane/linux/int_dataplane.go

```diff
@@ -607,49 +617,46 @@ func NewIntDataplaneDriver(config Config) *InternalDataplane {
 		)
 		dp.RegisterManager(failsafeMgr)
 
-		workloadIfaceRegex := regexp.MustCompile(strings.Join(interfaceRegexes, "|"))
-		bpfEndpointManager = newBPFEndpointManager(
-			&config,
-			fibLookupEnabled,
-			workloadIfaceRegex,
-			ipSetIDAllocator,
-			ipSetsMap,
-			stateMap,
-			ruleRenderer,
-			filterTableV4,
-			dp.reportHealth,
-			dp.loopSummarizer,
-		)
-		dp.RegisterManager(bpfEndpointManager)
-
 		// Pre-create the NAT maps so that later operations can assume access.
 		frontendMap := nat.FrontendMap(bpfMapContext)
+		frontendMap.SetMaxEntries(config.BPFMapSizeNATFE)
 		err = frontendMap.EnsureExists()
 		if err != nil {
 			log.WithError(err).Panic("Failed to create NAT frontend BPF map.")
 		}
+		maxEntriesMap[frontendMap.GetName()] = uint32(config.BPFMapSizeNATFE)
```

Comment:
You could add the max entries information to the `bpfMapContext` and pass that context object to bpfEndpointManager etc.  Then that could be the place where this config comes together.  I.e. you wouldn't need to do 
```
maxEntriesMap[frontendMap.GetName()] = uint32(config.BPFMapSizeNATFE)
```
here because that'd all be internal to the map context object.

Reply:
We use the map context for all of the maps, are you suggesting to place all the sizes in the context? I was wondering how to leverage the context, couldn't come up with something I particularly like :sweat_smile:

Reply:
I am thinking of doing this as part of the map infra cleanup. WDYT?